server/train/pipeline.py

```diff
@@ -49,10 +55,17 @@ def update_metadata(self, item):
         metadata_file = os.path.join(self.save_path, METADATA_FILENAME)
         with open(metadata_file, "w") as f:
             json.dump(item, f)
-        
+        if not self.weight_type:
+            self.archive_model()
+            
     @abstractmethod
     def train(self, prom_client):
         return NotImplemented
 
-    def get_model_specific_metadata(self):
-        return dict()
+    def archive_model(self):
+        archived_file = get_archived_file(self.group_path, self.model_name) 
+        print("achive model ", archived_file, self.save_path)
+        if os.path.exists(archived_file):
+            os.remove(archived_file)
```

Comment:
I do not think you need to remove the archived file. If it is saved you should be able to just call make_shutil and it should replace it. I will double check.